build_scripts/build_runner.py

```diff
@@ -622,7 +622,32 @@ def _copy(self):
         if build_state['status'] == "PASS":
             last_build_path = build_dir.relative_to(build_root_dir)
             last_build_file = build_dir.parent.parent / f'last_build_{self._component.build_info.product_type}'
-            last_build_file.write_text(str(last_build_path))
+            if self._is_latest_revision(last_build_file):
+                last_build_file.write_text(str(last_build_path))
+
+        return True
+
+    def _is_latest_revision(self, last_build_file):
+        """
+            Check that current revision is latest
+
+            :param last_build_file: Path to last_build_* file
+            :type last_build_file: pathlib.Path
+        """
+
+        try:
+            with last_build_file.open('r') as last_build_path:
+                manifest = Manifest(last_build_file.parent / last_build_path.read() / 'manifest.yml')
```

Comment:
Manifest(last_build_file.parent / last_build_path.read() / 'manifest.yml') generate wrong path to manifest.
Moreover, we cannot use manifests from artifact folders - rebuilds of old builds will be rewrite last successful build file.